packages/hardhat-core/test/internal/hardhat-network/stack-traces/test.ts

```diff
@@ -335,22 +340,15 @@ function compareStackTraces(
       `Stack trace of tx ${txIndex} entry ${i} type is incorrect: expected ${expectedErrorType}, got ${actualErrorType}`
     );
 
-    const actualMessage = (actual as any).message as
-      | ReturnData
-      | string
-      | undefined;
-
-    // actual.message is a ReturnData in revert errors, but a string
-    // in custom errors
-    let decodedMessage = "";
-    if (typeof actualMessage === "string") {
-      decodedMessage = actualMessage;
-    } else if (
-      actualMessage instanceof ReturnData &&
-      actualMessage.isErrorReturnData()
-    ) {
-      decodedMessage = actualMessage.decodeError();
-    }
+    // actual.message is a ReturnData in revert errors but in custom errors
+    // we need to decode it
+    const decodedMessage =
+      "message" in actual
+        ? actual.message
+        : "returnData" in actual &&
+          new ReturnData(actual.returnData).isErrorReturnData()
+        ? new ReturnData(actual.returnData).decodeError()
```

Comment:
This is a quibble, and not one I would block on, but I would be tempted to make `isErrorReturnData` a static, and pass `actual.returnData` in as an argument to that static method.

Reply:
I'm not sure if we can have static methods using napi-rs though. If we can, yes, I agree both with the idea and with doing it in a separate PR.

Btw, these tests will be removed in a follow-up PR, so even if we do it it will only show up in the EDR repo.

Reply:
In which case, this is fine.